index.mld

```diff
@@ -0,0 +1,97 @@
+{0 cwe_checker}
+
+{1 Contents}
+
+- {!section:Basics}
+- {!section:CmdLineOptions}
+- {!section:ToolIntegration}
+- {!section:HackingHowto}
+
+{1:Basics Basic usage}
+
+{i cwe_checker} is implemented as a plugin for the {{: https://github.com/BinaryAnalysisPlatform/bap} Binary Analysis Platform} (BAP).
+To use it, just run BAP with {i cwe_checker} as a pass:
+{[bap [BINARY_FILE] --pass=cwe_checker]}
+This runs all static analysis based checks.
+You can find more documentation on these checks at their {{: ../cwe_checker_core/Cwe_checker_core/index.html} module documentation pages}.
+The behaviour of these checks can be modified through a configuration file.
+Just edit the [src/config.json] file and then add [--cwe-checker-config=src/config.json] as a command line option.
+You can find more on the available command line options {{!section:CmdLineOptions} here}.
+
+The symbolic execution based checks can be run with the emulation recipe in the recipes folder.
+{[bap PATH_TO_BINARY --recipe=recipes/emulation]}
+
+{2 How to use the docker image}
+
+The docker image can be installed with
+{[docker pull fkiecad/cwe_checker]}
+To use it, mount the target binary inside the docker container and call {i bap} with {i cwe_checker} as a pass as usual:
+{[docker run --rm -v [BINARY]:/tmp/input fkiecad/cwe_checker bap /tmp/input --pass=cwe-checker]}
+If you are using a customized [config.json] file, don't forget to mount it inside your container as well!
+
+{1:CmdLineOptions Command line options}
+
+All command line options have to be prefixed with [--cwe-checker] (so that BAP knows to forward them to the {i cwe_checker} plugin).
```

Comment:
Note to us (maybe story): should we write a wrapper around bap (just a shellscript) and put it on the path. This wrapper would be called like cwe_checker --out BLA --json BINARY and it would translate it to bap BINARY --pass=cwe_checker --cwe-checker-out BLA --cwe-checker-json. This would made it more transparnet for the users!

Reply:
Yes, that would be useful, especially if combined with the docker container. I'll write a story for it.